src/Ocelot/LoadBalancer/LoadBalancers/LeastConnection.cs

```diff
@@ -126,7 +125,9 @@ private Response UpdateServices(List<Service> services)
 
                 foreach (var service in services)
                 {
-                    var exists = _leases.FirstOrDefault(l => l.HostAndPort.DownstreamHost == service.HostAndPort.DownstreamHost && l.HostAndPort.DownstreamPort == service.HostAndPort.DownstreamPort);
+                    var exists = _leases.FirstOrDefault(l => l.HostAndPort.DownstreamHost == service.HostAndPort.DownstreamHost
+                                    && l.HostAndPort.DownstreamPort == service.HostAndPort.DownstreamPort
+                                    && l.HostAndPort.Scheme == service.HostAndPort.Scheme);
```

Comment:
It's important to understand that we use equality `operator` definitions in both classes:
- [Lease](https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot/LoadBalancer/Lease.cs)
- [ServiceHostAndPort](https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot/Values/ServiceHostAndPort.cs)